atlasdb-perf/src/main/java/com/palantir/atlasdb/performance/benchmarks/KvsGetDynamicBenchmarks.java

```diff
@@ -0,0 +1,99 @@
+package com.palantir.atlasdb.performance.benchmarks;
+
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang.Validate;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import com.google.common.primitives.Ints;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.atlasdb.performance.backend.KeyValueServiceConnector;
+
+/**
+ * Performance benchmarks for KVS get with dynamic columns.
+ *
+ * @author coda
+ *
+ */
+@State(Scope.Thread)
+@BenchmarkMode(Mode.AverageTime)
```

Comment:
Cheers David, @rjullman, this Mode is probably worth us looking at when we consider the metrics we actually want to gather. I've just run Coda's tests with `@BenchmarkMode({Mode.AverageTime, Mode.SampleTime})` and this is the (abridged) results:
```
Benchmark                                         (backend)    Mode    Cnt      Score       Error  Units
KvsGetDynamicBenchmarks.getAllColumns              POSTGRES    avgt     20    872.483 ±    57.181  ms/op
KvsGetDynamicBenchmarks.getFirstColumnExplicitly   POSTGRES    avgt     20    285.921 ±    42.203  us/op
KvsGetDynamicBenchmarks.getAllColumns              POSTGRES  sample     40    917.976 ±    42.862  ms/op
KvsGetDynamicBenchmarks.getFirstColumnExplicitly   POSTGRES  sample  69670    290.495 ±    16.659  us/op
```
I'm doing to take a closer look at the [docs for the benchmark modes](http://hg.openjdk.java.net/code-tools/jmh/file/397891feda1b/jmh-samples/src/main/java/org/openjdk/jmh/samples/JMHSample_02_BenchmarkModes.java) and we should discuss.